# HG changeset patch
# Parent 7aeefc63c233dea88e3d3d7d1319c3eb1993075c
nsUpdateDriver.cpp: go up two directories on Windows/Linux due to different layout of appdir in Komodo


diff --git a/toolkit/xre/nsUpdateDriver.cpp b/toolkit/xre/nsUpdateDriver.cpp
--- a/toolkit/xre/nsUpdateDriver.cpp
+++ b/toolkit/xre/nsUpdateDriver.cpp
@@ -389,7 +389,9 @@ ApplyUpdate(nsIFile *greDir, nsIFile *up
   // Get the directory to which the update will be applied. On Mac OSX we need
   // to apply the update to the Foo.app directory which is the parent of the
   // parent of the appDir. On other platforms we will just apply to the appDir.
-#if defined(XP_MACOSX)
+  //
+  // ActiveState change: up two dirs on Windows and Linux as well because of
+  // Komodo's different install structure to Firefox.
   nsCAutoString applyToDir;
   {
     nsCOMPtr<nsIFile> parentDir1, parentDir2;
@@ -399,17 +401,14 @@ ApplyUpdate(nsIFile *greDir, nsIFile *up
     rv = parentDir1->GetParent(getter_AddRefs(parentDir2));
     if (NS_FAILED(rv))
       return;
-    rv = parentDir2->GetNativePath(applyToDir);
+    #if defined(XP_WIN)
+      nsAutoString applyToDirW;
+      rv = parentDir2->GetPath(applyToDirW);
+      CopyUTF16toUTF8(applyToDirW, applyToDir);
+    #else
+      rv = parentDir2->GetNativePath(applyToDir);
+    #endif
   }
-#elif defined(XP_WIN)
-  nsAutoString applyToDirW;
-  rv = appDir->GetPath(applyToDirW);
-
-  NS_ConvertUTF16toUTF8 applyToDir(applyToDirW);
-#else
-  nsCAutoString applyToDir;
-  rv = appDir->GetNativePath(applyToDir);
-#endif
   if (NS_FAILED(rv))
     return;
 
